### PR TITLE
Fix baselines computation

### PR DESCRIPTION
**Pull request by adamwawrzynski**
## Description

Refine docstrings, add batch baselines for batch input data processing, add possibility to provide multiple baselines for single batch and compute mean from all attributes.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [x] `pytest run tests`

## Checklist:

- [ ] I have updated all related `pyproject.toml` files.
- [x] I have updated lock files.
- [x] I have set an Assignee and Reviewers.
- [x] I have labeled the PR correctly.
- [x] I have performed a self-review of my own code.
- [x] I have covered my code and changes with unit tests.
- [x] I have annotated types as extensively as possible.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] All code references and dependencies will work.

### CODE REVIEW

File: foxai/explainer/computer_vision/algorithm/conductance.py [hunk 1]
```diff
@@ -173,16 +179,35 @@ def calculate_features(
             layer = get_last_conv_model_layer(model=model)
 
         conductance = self.create_explainer(model=model, layer=layer)
-        attributions = conductance.attribute(
-            input_data,
+
+        attributions_list: List[torch.Tensor] = []
```
Comment (joannasendorek) [inline foxai/explainer/computer_vision/algorithm/conductance.py:183]: You could use new AttrobutionType ;)

File: foxai/explainer/computer_vision/algorithm/gradient_shap.py [hunk 2]
```diff
@@ -165,39 +173,46 @@ def calculate_features(
...
-                input_data,
-                n_samples=n_samples,
-                stdevs=stdevs,
-                baselines=baselines,
-                target=pred_label_idx,
-                return_convergence_delta=False,
-                additional_forward_args=additional_forward_args,
-            )
-        validate_result(attributions=attributions)
+        attributions_list: List[torch.Tensor] = []
```
Comment (joannasendorek) [inline foxai/explainer/computer_vision/algorithm/gradient_shap.py:176]: And same here.

Comment (adamwawrzynski) [reply foxai/explainer/computer_vision/algorithm/gradient_shap.py]: Fixed: https://github.com/softwaremill/FoXAI/pull/97/commits/e62e2b2b737447cb080e63aedf5847054f1d3b1d.

File: foxai/explainer/computer_vision/algorithm/conductance.py (hunk 1 above)
Comment (adamwawrzynski) [reply foxai/explainer/computer_vision/algorithm/conductance.py]: Fixed: https://github.com/softwaremill/FoXAI/pull/97/commits/e62e2b2b737447cb080e63aedf5847054f1d3b1d.

File: foxai/explainer/computer_vision/model_utils.py [hunk 3]
```diff
@@ -47,3 +48,34 @@ def get_last_conv_model_layer(model: ModelType) -> LayerType:
...
+        baselines: Single tensor matching data sample dimensions or tensor
+            containing batch of baselines for each data sample.
+        input_data_shape: Shape of data sample batch.
+
+    Returns:
+        Tuple of baselines list and flag indicating if attributes should be
+            aggregated.
+    """
+    aggregate_attributes: bool = False
+    baselines_list: List[Union[None, torch.Tensor]] = [baselines]
```
Comment (joannasendorek) [inline foxai/explainer/computer_vision/model_utils.py:72]: BaselineType to be more consistent.

**Review by joannasendorek — APPROVED** [review]

File: foxai/explainer/computer_vision/model_utils.py (hunk 3 above)
Comment (adamwawrzynski) [reply foxai/explainer/computer_vision/model_utils.py]: Fixed: https://github.com/softwaremill/FoXAI/pull/97/commits/e526164d4b0d6a6b4108a458a75bd9e6e2728776.

**Pull request merged**